src/Database/LSMTree/Internal.hs

```diff
@@ -1172,7 +1172,7 @@ createSnapshot resolve snap label tableType t = do
           -- underfull buffer. However, note that this bit of code
           -- here is probably going to change anyway because of #392
           let credits = case confWriteBufferAlloc conf of
-                AllocNumEntries n -> Credits (unNumEntries n)
+                AllocNumEntries n -> Credits . fromEnum $ unNumEntries n
```

Comment:
I prefer `fromEnum` over `fromIntegral` when I *know* I am converting to an `Int` for for two reasons, neither of which are very substantial:
  1. `fromEnum` is shorter than `fromIntegral` leading to shorter lines.
  2. `fromEnum` imposes onlt the `Enum` constraint, whereas `fromIntegral` imposes *both* `Enum` and `Integral` (since `Enum` is a super-set of `Integral`).
  
That said, it could be argued that `fromIntegral` imposes less of a maintenance burden *just in case* the target type changes from `Int` to some other `Num` type.

Reply:
Do you know which of the two is faster? IIRC `fromEnum`/`toEnum` can fail with bounds checks